sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -229,36 +232,158 @@ protected virtual ValueTask<bool> IsEnvironmentReadyAsync()
+                return;
+            }
+
+            // intentionally not using the Credential property as we don't want to throw if the env vars are not available, and we want to allow this to vary per environment.
+            var credential = new ClientSecretCredential(
+                    tenantId,
+                    clientId,
+                    clientSecret,
+                    new ClientSecretCredentialOptions()
+                    {
+                        AuthorityHost = new Uri(authorityHost)
+                    });
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(ClientOptions.Default, new BearerTokenAuthenticationPolicy(credential, "https://management.azure.com/.default"));
 
-            throw new InvalidOperationException("The environment has not become ready, check your TestEnvironment.IsEnvironmentReady scenario.");
+            // create the GET request for the resource group information
+            Request request = pipeline.CreateRequest();
+            Uri uri = new Uri($"{ResourceManagerUrl}/subscriptions/{subscription}/resourcegroups/{resourceGroup}?api-version=2021-04-01");
+            request.Uri.Reset(uri);
+            request.Method = RequestMethod.Get;
+
+            // send the GET request
+            Response response = await pipeline.SendRequestAsync(request, CancellationToken.None);
+
+            // resource group not found - nothing we can do here
+            if (response.Status == 404)
+            {
+                return;
+            }
+
+            // unexpected response => throw an exception
+            if (response.Status != 200)
+            {
+                throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(response);
+            }
+
+            // parse the response
+            JsonElement json = JsonDocument.Parse(response.Content).RootElement;
+            if (json.TryGetProperty("tags", out JsonElement tags) && tags.TryGetProperty("DeleteAfter", out JsonElement deleteAfter))
+            {
+                DateTimeOffset deleteDto = DateTimeOffset.Parse(deleteAfter.GetString());
+                if (deleteDto.Subtract(DateTimeOffset.Now) < TimeSpan.FromDays(5))
+                {
+                    // construct the JSON to send for PATCH request
+                    using var stream = new MemoryStream();
+                    var writer = new Utf8JsonWriter(stream);
+                    writer.WriteStartObject();
+                    writer.WritePropertyName("tags");
+                    writer.WriteStartObject();
+
+                    // even though this is a PATCH operation, we still need to include all other tags
+                    // otherwise they will be deleted.
+                    foreach (JsonProperty property in tags.EnumerateObject())
+                    {
+                        if (property.NameEquals("DeleteAfter"))
+                        {
+                            DateTimeOffset newTime = deleteDto.AddDays(5);
```

Comment:
I think this is too long. It's already defaulted now to 5 days, but you're extending it another 5 days every time you access it? If you access it often, 2 or 3 days should be more than enough each time. The whole `DeleteAfter` change was meant to reduce costs.

@weshaggard @kurtzeborn ?

Reply:
It's being extended another 5 days only if the expiration is less than 5 days in the future.

Reply:
Why not make it Now + 5 days?

Reply:
My goal here was to make it so that the resource group wouldn't expire if you run live tests at least once a week.

Reply:
Now + 5 would at least be a little better, but if you're actively working on a feature still seems like a long time. I'll defer to @weshaggard and @kurtzeborn there.

Reply:
I'm not super worried about this and we can start here and then make adjustments if needed after some we see how it works for folks. It might also be worth adding an option to allow folks to configure this to be something different for some tests so that expensive resources could set this to a lower value.